### PR TITLE
Improving locators

### PR DESCRIPTION
**Pull request by luiCham**

### CODE REVIEW

File: src/page/bank-payment.page.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ export class BankPaymentPage {
   private confirmButton: ElementFinder;
 
   constructor () {
-    this.confirmButton = $('#cart_navigation > button > span');
+    this.confirmButton = $('#center_column #cart_navigation button[type="submit"]');
```
Comment (veronicatofino) [inline src/page/bank-payment.page.ts:7]: Why is this a better solution than the previous one?

File: src/page/payment-step.page.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ export class PaymentPage {
   private paymentButton: ElementFinder;
 
   constructor () {
-    this.paymentButton = $('#HOOK_PAYMENT > div:nth-child(1) > div > p > a');
+    this.paymentButton = $('#center_column .paiement_block #HOOK_PAYMENT .bankwire');
```
Comment (veronicatofino) [inline src/page/payment-step.page.ts:7]: You can improve more this solution

File: src/page/product-list.page.ts [hunk 3]
```diff
@@ -4,7 +4,7 @@ export class ProductListPage {
   private shirt: ElementFinder;
 
   constructor () {
-    this.shirt = $('#center_column a.button.ajax_add_to_cart_button.btn.btn-default');
+    this.shirt = $('#center_column ul.product_list > li:nth-child(1) a.button.ajax_add_to_cart_button.btn.btn-default');
```
Comment (veronicatofino) [inline src/page/product-list.page.ts:7]: You can improve more this solution

File: src/page/shipping-step.page.ts [hunk 4]
```diff
@@ -5,8 +5,8 @@ export class ShippingPage {
   private checkBox: ElementFinder;
 
   constructor () {
-    this.nextButton = $('#form > p > button > span');
-    this.checkBox =  $('#cgv');
+    this.nextButton = $('.cart_navigation button[name="processCarrier"]');
+    this.checkBox =  $('.order_carrier_content input#cgv');
```
Comment (veronicatofino) [inline src/page/shipping-step.page.ts:9]: Why is this a better solution?

**Review by veronicatofino — CHANGES_REQUESTED** [review]

File: src/page/shipping-step.page.ts (hunk 4 above)
Comment (veronicatofino) [reply src/page/shipping-step.page.ts:9]: I mean, why do you put .order_carrier_content input before #cgv when #cgv was already an unique locator?

File: src/page/payment-step.page.ts (hunk 2 above)
Comment (veronicatofino) [inline src/page/payment-step.page.ts]: You can improve more this solution

File: src/page/shipping-step.page.ts (hunk 4 above)
Comment (veronicatofino) [inline src/page/shipping-step.page.ts]: Why is this a better solution? I mean, why do you put .order_carrier_content input before #cgv when #cgv was already an unique locator?

Comment (luiCham) [reply src/page/shipping-step.page.ts]: I was thinking about improving speed through saving the selector some search time by giving him kind of a path to follow. After researching, i couldn't find anything about how the selectors deeply work that ensures it really saves searching time, plus i realized it wouldn't even be a notable improvement and it affects mantainability, so i'll just leave it as #cgv

File: src/page/sign-in-step.page.ts [hunk 5]
```diff
@@ -6,9 +6,9 @@ export class SignInPage {
   private passwordField: ElementFinder;
 
   constructor () {
-    this.nextButton = $('#SubmitLogin > span');
-    this.emailField = $('#email');
-    this.passwordField = $('#passwd');
+    this.nextButton = $('#login_form #SubmitLogin');
+    this.emailField = $('#login_form #email');
+    this.passwordField = $('#login_form #passwd');
   }
```
Comment (veronicatofino) [inline src/page/sign-in-step.page.ts:12]: Is the same in this cases. Just the id is enough.

**Review by veronicatofino — APPROVED** [review]

**Pull request merged**